core/modules/comment/comment.module

```diff
@@ -1165,6 +1160,20 @@ function comment_form_node_type_form_alter(&$form, $form_state) {
       '#default_value' => $node_type->settings['comment_close_days'],
       '#field_suffix' => t('days'),
     );
+    $form['comment']['titles'] = array(
+      '#markup' => t('<label>Comment titles</label>'),
+    );
+    $form['comment']['comment_subject_field'] = array(
+      '#type' => 'checkbox',
+      '#title' => t('Allow a custom title to be set for each comment'),
+      '#description' => t('When disabled, the comment title will be automatically generated from the comment text.'),
+    );
```

Comment:
This is missing the default value: `$node_type->settings['comment_subject_field']`